### PR TITLE
Work on probes and health checks.

### PR DESCRIPTION
**Pull request by geeknoid**
- Introduce probe functionality. Basic abstractions and k8s implementation.

- Streamline the APIs for health checks by removing needless overloads.

- Rename the HealthChecks.Core package to HealthChecks.Common

- Improve the health status produced by the application lifecycle health check to indicate why it is unhealthy.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/extensions/pull/4042)

### CODE REVIEW

File: src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Microsoft.Extensions.Diagnostics.ResourceMonitoring.csproj [hunk 1]
```diff
@@ -34,7 +34,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Bcl.TimeProvider" />
+    <PackageReference Include="Microsoft.Bcl.TimeProvider"/>
```
Comment (dbarbosapn) [inline src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Microsoft.Extensions.Diagnostics.ResourceMonitoring.csproj:37]: nit: was this whitespace removal intentional?

Comment (geeknoid) [reply src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Microsoft.Extensions.Diagnostics.ResourceMonitoring.csproj]: Fixed.

**Review by dbarbosapn — APPROVED** [review]
LGTM! :)

**Comment by RussKie** [comment]
@geeknoid @dbarbosapn  I missed this pull request, but essentially any changes in our package offering - i.e., an addition, a removal or name change of a package must be reflected in https://github.com/dotnet/api-catalog-infra/ so that the Docs team has the right content for the API docs. You can use https://github.com/dotnet/api-catalog-infra/pull/68 as a reference.

**Pull request merged**